### PR TITLE
WebRTC requestor cluster

### PR DESCRIPTION
**Pull request by andyg-apple**
WebRTC Requestor cluster

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
@andyg-apple I assume this is based on https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10004 ?

File: src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml:1]: Note spec review comments at https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10004#pullrequestreview-2349495429 (though those don't affect this XML).

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    <command code="0x01" source="client" name="Offer" optional="false">
+      <description>This command provides the stream requestor with WebRTC session details.</description>
+      <arg id="1" name="WebRTCSessionID" type="int16u"/>
+      <arg id="2" name="SDP" type="char_string"/>
+      <arg id="3" name="ICEServers" array="true" type="ICEServerStruct" optional="true"/>
+      <arg id="4" name="ICETransportPolicy" type="char_string" optional="true"/>
+    </command>
+
+    <command code="0x02" source="client" name="Answer" optional="false">
+      <description>This command provides the stream requestor with the WebRTC session details (i.e.</description>
```
[the remark this answers is not in the document]
Comment (andyg-apple) [reply src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml]: Thanks, have done so. Looks like command descriptions from Alchemy are not fully reliable rn.

**Review by pidarped — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <domain>Cameras</domain>
+    <name>WebRTC Transport Requestor</name>
+    <code>0x0554</code>
+    <define>WEBRTC_TRANSPORT_REQUESTOR_CLUSTER</define>
+    <description>The WebRTC transport requestor cluster provides a way for stream consumers (e.g. Matter Stream Viewer) to establish a WebRTC connection with a stream provider.</description>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <globalAttribute code="0xFFFD" side="either" value="1"/>
+    <attribute code="0x0000" side="server" define="CURRENT_SESSIONS" type="array" entryType="WebRTCSessionStruct" default="0">CurrentSessions</attribute>
+    <command code="0x01" source="client" name="Offer" optional="false">
```
Comment (pidarped) [inline src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml:35]: We need support in Alchemy for Large message quality tag in xml for commands.
Filed https://github.com/project-chip/alchemy/issues/11.

File: scripts/tools/zap/tests/outputs/all-clusters-app/app-templates/endpoint_config.h
[the remark this answers is not in the document]
Comment (andyg-apple) [reply scripts/tools/zap/tests/outputs/all-clusters-app/app-templates/endpoint_config.h:1]: Thanks, had to nuke my local rep in the end...

File: src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml (hunk 2 above)
Comment (andyg-apple) [reply src/app/zap-templates/zcl/data-model/chip/webrtc-requestor-cluster.xml:35]: Filed https://github.com/project-chip/connectedhomeip/issues/35951 top add this later.

**Pull request merged**